pkg/controllers/vdb/readonlyonlineupgrade_reconciler.go

```
@@ -454,17 +457,59 @@ func (o *ReadOnlyOnlineUpgradeReconciler) drainSubcluster(ctx context.Context, s
 		return ctrl.Result{}, err
 	}
 
-	if img != o.Vdb.Spec.Image {
-		scName := sts.Labels[vmeta.SubclusterNameLabel]
-		o.Log.Info("rerouting client traffic from subcluster", "name", scName)
-		if err := o.routeClientTraffic(ctx, scName, true); err != nil {
-			return ctrl.Result{}, err
-		}
+	if img == o.Vdb.Spec.Image {
+		return ctrl.Result{}, nil
+	}
+	scName := sts.Labels[vmeta.SubclusterNameLabel]
+	o.Log.Info("rerouting client traffic from subcluster", "name", scName)
+	if err := o.routeClientTraffic(ctx, scName, true); err != nil {
+		return ctrl.Result{}, nil
+	}
+
+	o.Log.Info("starting check for active connections in subcluster", "name", scName)
+	return o.runSubclusterDrain(ctx, scName)
+}
+
+func (o *ReadOnlyOnlineUpgradeReconciler) runSubclusterDrain(ctx context.Context, scName string) (ctrl.Result, error) {
+	drainSubclusterAnnotation := "vertica.com/drain-start-time-" + scName
+	pfs := o.PFacts.FindPodsBySubcluster(scName)
 
-		o.Log.Info("starting check for active connections in subcluster", "name", scName)
-		return o.Manager.isSubclusterIdle(ctx, o.PFacts, scName)
+	// If no active connections, return
+	if res, err := o.Manager.isSubclusterIdle(ctx, o.PFacts, scName); !verrors.IsReconcileAborted(res, err) {
```

If err != nil, we want to let the user know so we should return the error